src/proxy/inbound.rs

```diff
@@ -73,8 +74,16 @@ impl Inbound {
 
     pub(super) async fn run(self) {
         let (tx, rx) = oneshot::channel();
-        let service =
-            make_service_fn(|_| async { Ok::<_, hyper::Error>(service_fn(Self::serve_connect)) });
+        let service = make_service_fn(|socket: &tokio_boring::SslStream<tokio::net::TcpStream>| {
+            let peer_addr = if self.cfg.enable_original_source {
```

Comment:
this seems a bit unexpected to me, I would expect `enable_original_source` to maybe only be checked in freebind_connect? I guess that would require passing `cfg` all the way through?